src/pages/member/UserEvents.js

```diff
@@ -0,0 +1,292 @@
+import React, { useState, useMemo, useEffect } from 'react'
+import { useHistory } from 'react-router-dom'
+import { connect } from 'react-redux'
+import { Helmet } from 'react-helmet'
+
+import EventCard from '../../components/Cards/Event'
+import { fetchBackend, updateEvents, updateRegisteredEvents } from '../../utils'
+
+import { setUser } from '../../actions/UserActions'
+
+import { COLOR } from '../../constants/Constants'
+import { withStyles } from '@material-ui/core/styles'
+import {
+  Tabs,
+  Tab,
+  List,
+  ListItem,
+  ListItemText,
+  Typography
+} from '@material-ui/core'
+
+import {
+  StarBorder,
+  PlaylistAddCheck,
+  Search
+} from '@material-ui/icons'
+
+// States for the filters
+const PANEL_STATES = {
+  FAVORITES: 'FAVORITES',
+  REGISTERED: 'REGISTERED',
+  ALL: 'ALL'
+}
+
+const TAB_STATES = {
+  UPCOMING: 0,
+  PAST: 1,
+  ALL: 2
+}
+
+const styles = ({
+  header: {
+    color: COLOR.BIZTECH_GREEN
+  },
+  flex: {
+    display: 'flex',
+    flexDirection: 'row',
+    paddingTop: '14px',
+    paddingBottom: '14px'
+  },
+  sidePanel: {
+    layout: {
+      display: 'flex',
+      float: 'right',
+      flexDirection: 'column',
+      textAlign: 'right',
+      marginRight: '3em',
+      minWidth: '15em'
+    },
+    title: {
+      fontSize: '3em'
+    },
+    button: {
+      textAlign: 'right'
+    },
+    activeButton: {
+      textAlign: 'right',
+      borderRight: `2px solid ${COLOR.BIZTECH_GREEN}`
+    }
+  },
+  tabs: {
+    container: {
+      width: '100%'
+    },
+    layout: {
+      marginBottom: '2em'
+    },
+    tab: {
+      fontSize: '0.9rem',
+      marginLeft: '1em',
+      marginRight: '1em',
+      textTransform: 'none',
+      maxWidth: '5em',
+      width: '100%',
+      color: COLOR.WHITE
+    }
+  },
+  rows: {
+    display: 'flex',
+    flexWrap: 'wrap'
+  }
+})
+
+function EventPanel (props) {
+  const { children, currentIndex, index, ...rest } = props
+
+  return (
+    <div
+      role='tabpanel'
+      hidden={currentIndex !== index}
+      id={`tabpanel-${index}`}
+      { ...rest }
+    >
+      {children}
+    </div>
+  )
+}
+
+function UserEvents (props) {
+  const [tabIndex, setTabIndex] = useState(TAB_STATES.ALL)
+  const [selectedPanel, setSelectedPanel] = useState(PANEL_STATES.ALL)
+
+  const history = useHistory()
+
+  const { events = [], eventsRegistered = [], user } = props
+
+  useEffect(() => {
+    if (!props.events) updateEvents()
+    if (user.id) updateRegisteredEvents(user.id)
+  }, [])
+
+  const handleFavoriteEvent = async (eventId, toggle) => {
+    const body = { eventID: eventId, isFavourite: toggle }
+    await fetchBackend(`/users/favEvent/${user.id}`, 'PATCH', body)
+    const newEventsId = toggle
+      ? [...user.favedEventsID, eventId]
+      : user.favedEventsID.filter((id) => id !== eventId)
+    props.setUser({
+      ...user,
+      favedEventsID: newEventsId
+    })
+  }
+
+  const handleTabChange = (event, newIndex) => {
```

Comment:
nit: unused param?

Reply:
The material.ui function automatically passes the first `event` argument which we don't really need.. I'm not sure if there is a way to make this look any cleaner, is there?

Reply:
oh, gotcha